### ISSUE TITLE
"oil create" installs "1.2/develop"

### ISSUE THREAD

**Issue by kenjis** [fuel/core#943]
Sould it install current released version, that is "1.1/master"?

**Comment by WanWizard** [comment]
This is a known issue and a bit of a catch-22.

Oil basically clones the default repository on github, as it doesn't have a way of knowing what the latest master is. And we can't set the default to the latest master, because we've seen that people started forking that, and sending pull requests for issues already fixed in the latest development branch.

We're busy figuring out a way to check what the latest master is, but haven't found a solution yet.

On the other hand, it does a 'git clone', so you can switch back to the latest master using
`git checkout <branch>` and `git submodule foreach git checkout <branch>`.

**Comment by kenjis** [comment]
Then, how about using branch name just "master", not "1.1/master"?
I don't see the merit of current branch naming rule. Why don't you use just "master" and "develop"?

**Comment by WanWizard** [comment]
Because we need to keep multiple versions. After the release of 1.1 a bit of a cockup was made, which we're going to correct once 1.2 is released and we will end up with 1.1/master (last release), 1.2/master (current release) 1.2/develop (for release fixes), and 1.3/develop (current development). Other branches will be removed.

**Comment by WanWizard** [comment]
Replace /usr/bin/oil with this script:

```
#!/bin/bash

if [ -f "./oil" ]; then
        php oil "$@"
else

        if [ "$1" == "create" ]; then

                                if [ ! `which git` ]; then 
                                        echo "For this installer to work you'll need to install Git."
                        echo '        http://git-scm.com/'
                                fi

                git clone --recursive git://github.com/fuel/fuel.git "./$2"

                cd ./$2
                branch=`git branch -a | grep "master" | tail -1 | cut -d/ -f3-4`
                git checkout $branch
                git submodule foreach git checkout $branch
                cd ..

                php "./$2/oil" refine install
        else
                echo 'This is not a valid Fuel installation so Oil is a bit lost.'
                echo '        http://fuelphp.com/docs/installation/instructions.html'

        fi
fi
```

@philsturgeon can you put this online if and when possible?

**Comment by philsturgeon** [comment]
It's pushed to GitHub, no idea how to get it to live.

Emailing from my iPhone 4S like a BOSS!

**Comment by WanWizard** [comment]
Which repo?

it still needs to be improved, ideally it needs to grab the correct branch before cloning, so the current submodule references for the cloned version can be used. Haven't got a clue how to do that atm.

**Comment by philsturgeon** [comment]
There is a special repo for it: 

https://github.com/fuel/installer 

## 

Phil Sturgeon

**Comment by WanWizard** [comment]
Ok. thanks. Learning all the time... ;)

For now, can you bump the /current-version to 1.2 when you have the chance?

**Comment by kenjis** [comment]
@WanWizard "oil create" should install current released version, I mean now not 1.2RC1 but 1.1, shouldn't it?

How about consider to change to a branching model like "A successful Git branching model"?
http://nvie.com/posts/a-successful-git-branching-model/

The point is "master" is always latest released version, and "develop" is always edge of development.

Every time to change the development branch name is a bother. :-)

**Comment by philsturgeon** [comment]
We already use Git Flow, this is an adaption of that which lets us work on multiple versions in tandem, without needing totally different repos (which would be silly) 

## 

Phil Sturgeon

**Comment by WanWizard** [comment]
@kenjis we have decided on this switch because 1.2RC1 is an awful lot more stable then 1.1. 

And it's only for a week or so, if no major issues come to light I plan to release 1.2 as soon as possible.

**Comment by kenjis** [comment]
I agree that 1.2RC1 is more stable than 1.1.

BTW, latest oil command was updated to install current versions master:
https://github.com/fuel/installer/commit/29a0110e3841049a309b731369aa90f4a7aa8ff3

But "http://get.fuelphp.com/installer.sh" is not updated.

**Comment by WanWizard** [comment]
We don't have access to both DNS and Pagodabox at the moment (Dan holds the accounts), we're working on resolving that issue.

**Comment by WanWizard** [comment]
This should be fixed by now, it currently installs 1.2/master, as it should.
